sktime/decomposition/vampnet.py

```diff
@@ -0,0 +1,662 @@
+
+    def fit(self, data, n_epochs=1, batch_size=512, validation_data=None,
+            train_score_callback: Callable[[int, torch.Tensor], None] = None,
+            validation_score_callback: Callable[[int, torch.Tensor], None] = None, **kwargs):
+        r""" Fits a VampNet on data.
+
+        Parameters
+        ----------
+        data : torch.Tensor or Dataset or TimelaggedDataset or numpy array
+            The data to use for training.
+        n_epochs : int, default=1
+            The number of epochs (i.e., passes through the training data) to use for training.
+        batch_size : int, default=512
+            The batch size to use during training. Should be reasonably large so that covariance matrices have
+            enough statistics to be estimated.
+        validation_data : torch.Tensor or Dataset or TimeLaggedDataset or numpy array, optional, default=None
+            Validation data.
+        train_score_callback : callable, optional, default=None
+            Callback function which is invoked after each batch and gets as arguments the current training step
+            as well as the score (as torch Tensor).
+        validation_score_callback : callable, optional, default=None
+            Callback function for validation data. Is invoked after each epoch if validation data is given
+            and the callback function is not None. Same as the train callback, this gets the 'step' as well as
+            the score.
+        **kwargs
+            Optional keyword arguments for scikit-learn compatibility
+
+        Returns
+        -------
+        self : VAMPNet
+            Reference to self.
+        """
+        self._step = 0
+
+        # set up loaders
+        data_loader = self._set_up_data_loader(data, batch_size=batch_size, shuffle=self.shuffle)
+        validation_loader = None
+        if validation_data is not None:
+            validation_loader = self._set_up_data_loader(validation_data, batch_size=batch_size, shuffle=False)
+
+        # and train
+        for epoch in range(n_epochs):
+            for batch_0, batch_t in data_loader:
+                self.partial_fit((batch_0.to(device=self.device), batch_t.to(device=self.device)),
+                                 train_score_callback=train_score_callback)
+            if validation_loader is not None and validation_score_callback is not None:
+                with torch.no_grad():
+                    scores = []
+                    for val_batch in validation_loader:
+                        scores.append(
+                            self.validate((val_batch[0].to(device=self.device), val_batch[1].to(device=self.device)))
+                        )
+                    mean_score = torch.mean(torch.stack(scores))
+                    validation_score_callback(self._step, mean_score)
+        return self
+
+    def transform(self, data, **kwargs):
+        r""" Transforms data through the instantaneous network lobe.
```

Comment:
is there an option to transform through the other lobe? do we need it?

Reply:
Good point. Didn't need so far but for sure is needed if timelagged data =/= instantaneous data (in terms of observables)